### PR TITLE
action menu button

### PR DESCRIPTION
**Pull request by oriv**

### CODE REVIEW

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  () => {
+    return {
+      template: storyTemplate,
+      props: {
+        menuItems: object('menuItems', menuItemsMock),
+        openLeft: boolean('openLeft', false),
+      },
+      moduleMetadata: {
+        imports: [
+          BrowserAnimationsModule,
```
Comment (tomeraronovsky) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts]: I think you don't need this module

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+})
+export class ActionsMenuButtonComponent implements OnInit {
+  @Input() menuItems: MenuItem[];
+  @Input() openLeft: boolean;
+  @Input() data: any;
+  readonly buttonType: ButtonType = ButtonType.tertiary;
+  readonly buttonColor: IconColor = IconColor.normal;
+  readonly buttonIcon: Icons = Icons.three_dots_vert;
+
+  constructor() { }
```
Comment (tomeraronovsky) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts]: remove if empty

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts (hunk 1 above)
Comment (galloween) [reply projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts]: it is needed for the menu to work in storybook. in the real app its imported in the app.module

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  templateUrl: './action-menu-button.component.html',
+  styleUrls: ['./action-menu-button.component.scss'],
+})
+export class ActionsMenuButtonComponent implements OnInit {
+  @Input() menuItems: MenuItem[];
+  @Input() openLeft: boolean;
+  @Input() data: any;
+  readonly buttonType: ButtonType = ButtonType.tertiary;
+  readonly buttonColor: IconColor = IconColor.normal;
+  readonly buttonIcon: Icons = Icons.three_dots_vert;
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts]: to make it more generic, maybe this should be inputs with default values, or maybe single [config] input with these props

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+@Component({
+  selector: 'b-action-menu-button',
+  templateUrl: './action-menu-button.component.html',
+  styleUrls: ['./action-menu-button.component.scss'],
+})
+export class ActionsMenuButtonComponent implements OnInit {
+  @Input() menuItems: MenuItem[];
+  @Input() openLeft: boolean;
+  @Input() data: any;
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts]: dont think this input is needed. when you create the menuItems config and menu actions, you have all the data there. this component should not know data.

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  ngOnInit() {
+    this.menuItems = map(this.menuItems, (item: MenuItem) => {
+      return assign({}, item, {
+        action: () => {
+          item.action(this.data);
+        }
+      });
+    });
+  }
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts]: not needed. when you generate the menuItems, you can bind anything there.

also, you might not need menu actions at all. using (actionClicked) event of the menu and providing correct id and key, all menu can be handled with just 1 method. 
Checkout the **note** at the bottom of this page (updated recently): https://hibobio.github.io/bob-style/?path=/info/components-navigation--menu

for this approach to work, this component should forward (actionClicked) event from the menu

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div class="b-action-menu-button">
+  <b-menu [menu]="menuItems"
+          [openLeft]="openLeft"
+          (click)="$event.stopPropagation()">
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.html]: should forward  (actionClicked) event from the menu

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+const story = storiesOf(ComponentGroupType.Buttons, module).addDecorator(
+  withKnobs
+);
+
+const template = `
+  <b-action-menu-button
+            [menuItems]="menuItems"
+            [openLeft]="openLeft">
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts]: add  (actionClicked)

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts (hunk 1 above)
Comment (galloween) [reply projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.stories.ts]: BrowserAnimationsModule  is needed for the menu to work in storybook

File: projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import { Component, Input, OnInit } from '@angular/core';
+import { assign, map } from 'lodash';
+import { MenuItem } from '../../navigation/menu/menu.interface';
+
+@Component({
+  selector: 'b-action-menu-button',
+  templateUrl: './action-menu-button.component.html',
+  styleUrls: ['./action-menu-button.component.scss'],
+})
+export class ActionsMenuButtonComponent implements OnInit {
```
Comment (galloween) [inline projects/ui-framework/src/lib/buttons/action-menu-button/action-menu-button.component.ts]: action or actionS ?  b-action-menu-button but  ActionsMenuButtonComponent...

**Pull request merged**